### DISCUSSION TITLE
Cannot display cards in page

### DISCUSSION THREAD

**Discussion by aplasa85** [neuefische/questions#390] [Web] [answered]
Hi everybody. I am having some difficulties to render the product details of my tequila and mezcal app. As a matter of fact, I am trying to somehow fetch the info of the array information to a details page with just one product, but have so far not been quite as successful. If someone could give me a piece of advise on how to solve this, I would be thankful.

return (
    <>
      <Link href="/">
        <StyledTitle>Casa del Agave</StyledTitle>
      </Link>

      async function fetchProducts(product) {
     response = await fetchProducts (product)
     const tequilaData = await TequilaList.json
      return tequilaData.results;

    tequilaData.map(tequila =
    <>
      {Tequila.brand}
      {Tequila.rating}
      {Tequila.price}
      {Tequila.image}
      </>
    </>
  );
      }

export default 
);

    https://github.com/aplasa85/capstone-template-ffm-web-22-1

**Comment by stefankorder** [comment]
Hi Axel,
on which page are you trying to fetch something? I can't find the location you want to fetch.

**Comment by aplasa85** [comment]
I am fetching from products/[ProductId].js.

**Comment by mihaimazareanu** [comment]
I noticed that there are some mistakes in your map function.

You have written this: 

    tequilaData.map(tequila =
    <>
    {Tequila.brand}
    {Tequila.rating}
    {Tequila.price}
    {Tequila.image}
    </>
    </>
    );
   }
   
   You are missing one > in your arrow function, then you map `tequilaData` and use `tequila` as parameter, but then you render `Tequila.brand` and so on, instead of `tequila.brand` And then you have 2 `</>` before you close your map function.
   
   So I think your map function should look like this:
   
    tequilaData.map(tequila =>
    <>
    {tequila.brand}
    {tequila.rating}
    {tequila.price}
    {tequila.image}
    </>
    );

I hope this helps.

**Comment by aplasa85** [comment, marked as answer]
Did try it out, and it did the trick indeed. Nonetheless, it seems the whole task has been changed since then, thus the problem is no longer pending. Thanks a lot Mihai.
